### ISSUE TITLE
Access denied after updating to 1.8.1

### ISSUE THREAD

**Issue by norwayRP** [overextended/oxmysql#70]
## Issue checklist
> Please put `x` inside of the box that matches your issue.

- [x ] I am using the latest release
- [x] I have referenced previously reported issues
- [x ] I have referenced available documentation and cannot resolve the issue
- [x ] I'm certain this is an issue with oxymysql, and not with my resource

**Describe the bug**
After updating to latest I get
`Access denied for user 'root'@'localhost' (using password: YES)`
Not showing the correct IP as I have in cfg file. Previous version works fine.
**Screenshots**

**Server details**
- FX Server artifact : 4978
- Operating system: Linux
- Framework: QB-Core

**Comment by thelindat** [comment]
Show your connection string.

**Comment by norwayRP** [comment]
`set mysql_connection_string "mysql://root:password@198.80.59.86/nrp35?charset=utf8mb4"`

**Comment by dunak-debug** [comment]
Is your password "password"?

**Comment by norwayRP** [comment]
> Is your password "password"?

nope, whats the relevanse. Not sharing my info here anyways.

**Comment by thelindat** [comment]
> Not showing the correct IP

So you're entering the IP above and getting something else entirely? Have you got some other resource setting the connection string, is the database not running locally (in which case use localhost)? Somebody had this issue with a docker container IIRC at one point.

**Comment by norwayRP** [comment]
Lets say I am using that IP, but the error show a diffrent IP adress. I do not have any resources setting a connection string.
Moved back to 1.8.0 and all worked fine again.
![image](https://user-images.githubusercontent.com/45201861/144032402-b2269088-5943-4069-951c-83aec16c8933.png)

This is whats showing x all my scripts connecting to sql.

**Comment by thelindat** [comment]
172.18.* addresses are private network addresses (similar to 192.168.*). URI parsing is handled by the node module (mysql2) and not us, and it's not something being reported there.
I'll ask again, are the server and database hosted separately or are you using docker? Any other relevant information about your server setup?

**Comment by norwayRP** [comment]
The Server and db are on same machine. I am using linux with  Pterodactyl panel for easy control. I am using my public IP to connect with db.

**Comment by thelindat** [comment]
Half the issues people have are related to Pterodactyl, and I assume it needs some specific settings to be used.

If you're running both mysql and fxserver inside the container, can you not just connect with localhost (not the public IP)?

**Comment by norwayRP** [comment]
Never had any issues with Pterodactyl before, but why using public IP instead of local host is security related I was told by I friend that helped me set it up. I am not that good with sql stuff.

But whats the big change between 1.8.0 and 1.8.1?

**Comment by dunak-debug** [comment]
IP is not problem, he can connect to database. Access denied means that his user / password are wrong

**Comment by norwayRP** [comment]
> IP is not problem, he can connect to database. Access denied means that his user / password are wrong

NO, username and password is correct. As I said using previous version works fine then the username and password can not be wrong.

**Comment by thelindat** [comment]
> username and password can not be wrong.

Does your password contain a special character such as `@`, that you previously replaced with `%40`?

**Comment by norwayRP** [comment]
> 

Ah, password characters might be the thing then yes. Let me check.

**Comment by norwayRP** [comment]
Thanks @thelindat , that was it. Did not think of that.

**Comment by thelindat** [comment]
> Thanks @thelindat , that was it. Did not think of that.

Neither did I, else I wouldn't have marked 1.8.1 as a patch rather than minor.
